### PR TITLE
feat: add recipe for upgrading to `slack-github-action` v2

### PR DESCRIPTION
**Pull request by sjungling**
featured breaking changes in how messages were posted to slack. this handles the second scenario
defined in their repository.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -10,3 +10,12 @@ dependencies {
...
 }
+java {
+    toolchain {
+        languageVersion.set(JavaLanguageVersion.of(17))
+    }
+}
+
+tasks.named<JavaCompile>("compileJava") {
+    options.release.set(8)
+}
```
Comment (timtebeek) [inline build.gradle.kts:21]: Why these changes? Should be set through the plugin already right?
```suggestion
```

**Review by timtebeek — APPROVED** [review]
Neat to see! Approved already, but ideally we remove the toolchain and compile task overrides.

File: build.gradle.kts (hunk 1 above)
Comment (sjungling) [reply build.gradle.kts:21]: I can try revert this but  at the my IDE was having issues with the syntax.

**Review by timtebeek — APPROVED** [review]
